tsdb/querier/querier.go

```diff
@@ -95,11 +98,26 @@ func (q *Query) ExecWithoutEncode() QueryResponse {
 	return data
 }
 
-func (q *Query) exec(blockStream []tsdb.Block, encoding bool) interface{} {
+func (q *Query) exec(blockStream []tsdb.Block, encoding bool) (i interface{}) {
 	// start represents the starting of the timer that will calculate
 	// the total time involved in performing a particular query.
 	// This can be later benchmarked and compared with other algorithms.
 	base, stamp := getBaseResponse(q.Range)
+	defer func() {
+		var res QueryResponse
+		if q.encoding {
+			err := json.Unmarshal(i.([]byte), &res)
+			if err != nil {
+				log.Errorln(fmt.Errorf("Decoding error: %s", err.Error()).Error())
+			}
+			res.EvaluationTime = time.Since(stamp).String()
+			i, _ = json.Marshal(res)
```

Comment:
```suggestion
			if i, ok = json.Marshal(res); !ok {
			    log.Error("invalid type received")
			}
```

we should log error if not ok is false